docs/tutorial/typesfuns.rst

```diff
@@ -1182,3 +1183,30 @@ matching ``let`` and lambda bindings. It will *only* work if:
 - The type of the result is "known". i.e. the type of the expression
   can be determined *without* type checking the ``case``-expression
   itself.
+
+Totality
+========
+
+A *total* function is one that is defined for all inputs of the correct
+type: that is, it is guaranteed not to crash. For example, a function
```

Comment:
Maybe it should be something of the ilk of:

`it is guaranteed to return a valid result without diverging` or something like that you find suitable.